projects/web-wallet/rpc-protocol.md

```diff
@@ -0,0 +1,432 @@
+| -32053 | INVALID_PARAMETERS | Invalid parameters |
+| -32099     | GENERIC_FAILURE | Generic failure |
+
+# 3. Guide-level Explanation
+
+### 3.1 `htr_sendTx`
+
+When the request is received, we will validate if the parameters are valid, enforcing the data types and rules described in the Reference-Level explanation
+
+If the validation fails, we will respond with a `INVALID_PARAMETERS` error code
+
+If the validation is successful, we will display a confirmation screen, with all outputs for manual user validation (UX is yet to be discussed). If the user rejects the transaction, we will respond with a `REQUEST_REJECTED` error
+
+If the user accepts the transaction, we will create a `SendTransaction` model (or `SendTransactionWalletService` if on the wallet-service) and execute the send transaction.
+
+Since `JSON-RPC` is designed considering only a single response per request, we will only respond the request when the transaction either failed to be pushed to the fullnode (or the wallet-service) or when it's successfully pushed.
+
+## 3.2 `htr_createToken`
+
+When the request is received, we will validate if the parameters are valid by following the parameter type definitions in the Reference-Level explanation
+
+If the validation fails, we will respond with a `INVALID_PARAMETERS` error code
+
+At this point, we will call `prepareCreateNewToken` and display the assembled transaction for user validation (UX is yet to be discussed).
+
+If the user accepts the transaction, we will, just like the `htr_sendTx` method, create a `SendTransaction` model and execute the send transaction. Otherwise, we will answer with a `REQUEST_REJECTED` error
+
+## 3.3 `htr_getUtxos`
+
+When the request is received, we will validate if the parameters are valid by following the parameter type definitions in the Reference-Level explanation
+
+If the validation fails, we will respond with a `INVALID_PARAMETERS` error code
+
+At this point, we will call the `getUtxos` method to retrieve `utxos` that fulfill the requested filters.
+
+Before returning, we will display a confirmation screen (UX yet to be discussed) asking for user confirmation by displaying the `utxos` that will be returned to the `dApp`.
+
+If the user accepts the transaction, we will reply with the array of `utxos`. Otherwise, we will answer with a `REQUEST_REJECTED` error
+
+#### Future possibilities
+
+We might display "checkboxes" before each utxos so the user can choose exactly which ones he want to display
+
+## 3.4 `htr_signWithAddress`
+
+When the request is received, we will validate if the parameters are valid by following the parameter type definitions in the Reference-Level explanation
+
+If the validation fails, we will respond with a `INVALID_PARAMETERS` error code
+
+At this point, we will display the `message` and the `address` to the user, asking for his confirmation. If he rejects the modal, we will respond with an `INVALID_PARAMETERS` error code.
```

Comment:
I think we should prefix the message before showing it to the user. If the original message is already prefixed, we can skip it.

Reply:
Done in f8c0b94cf09a39c35d648a0850ae201593ec0ec8

Reply:
Thanks!